src/main/java/com/redhat/pantheon/servlet/AsciidocRenderingServlet.java

```diff
@@ -105,16 +88,18 @@ protected void doGet(SlingHttpServletRequest request,
         w.write(html);
     }
 
-    private boolean generatedContentHashDoesntMatch(Resource resource) {
-        String srcContent = resource.getValueMap().get("jcr:content/jcr:data", String.class);
-        String existingHash = resource.getValueMap().get("pant:cachedContent/pant:hash", String.class);
+    private boolean generatedContentHashMatches(Resource resource) {
+        String srcContent = resource.getValueMap().get("asciidoc/jcr:content/jcr:data", String.class);
```

Comment:
I guess this assumes the resource itself is not a file, but instead has an `asciidoc` node under it which is of type `nt:file`?

Reply:
Yeah, I had to rearrange the data structure a little bit due to the way file upload works in sling+HTL. The OOTB widget creates a node to hold most of the properties, with another child node underneath for the file. You can hint at what some of the more core property values should be, but sling can get confused about what node to apply the hint to.

For example, I tried to specify pant:module instead of the default nt:file, but then both the parent node and the file node became pant:module, which was bad. I didn't have a good reason to fight with the OOTB behavior, so I chose to make some simple changes to our data structure instead.

Reply:
I had the exact same problem, so I know what you are referring to. I finally made it work so I can share those findings with you. Having said that, this is a good approach and I can see the possibilities. Lets catchup offline on the pros/cons of this structure.